pulp_2to3_migration/app/pre_migration.py

```diff
@@ -623,12 +632,17 @@ def delete_old_resources(plan):
     with transaction.atomic():
         pulp2distributors_with_old_distributions_qs = Pulp2Distributor.objects.filter(
             old_dist_query).only('pulp3_distribution')
-
         for pulp2distributor in pulp2distributors_with_old_distributions_qs:
             if pulp2distributor.is_migrated:
                 pulp2distributor.is_migrated = False
                 pulp2distributor.save()
             if pulp2distributor.pulp3_publication:
+                # todo check when publication is shared by multiple distro
+                pulp2d=pulp2distributor.pulp3_publication.pulp2distributor_set.all()
+                for d in pulp2d:
+                    d.is_migrated=False
+                    d.save()
+                    d.pulp3_distribution.delete()
                 pulp2distributor.pulp3_publication.delete()
-            if pulp2distributor.pulp3_distribution:
+            elif hasattr(pulp2distributor, 'pulp3_distribution') and pulp2distributor.pulp3_distribution:
```

Comment:
There are cases when publication is not changing but distributor needs to be changed. That's why it was 2 separate `if`s.

I also suggest to collect all the distribution pks and remove them in bulk, they all should be of BaseDistribution type. What do you think?